src/internal/group_api/mod.rs

```diff
@@ -360,90 +450,94 @@ pub(crate) fn get_group_keys<'a>(
-                        // when we expand this to include admins
-                        let member_info_result: Result<
-                            HashMap<UserId, (PublicKey, Option<TransformKey>)>,
-                            _,
-                        > = member_ids_and_keys
+
+                        let group_members: Vec<requests::GroupMember> = member_info
+                            .into_iter()
+                            .map(|(member_id, member_pub_key, member_trans_key)| {
+                                requests::GroupMember {
+                                    user_id: member_id,
+                                    transform_key: member_trans_key.into(),
+                                    user_master_public_key: member_pub_key.into(),
+                                }
+                            })
+                            .collect();
+                        let maybe_group_members = if group_members.is_empty() {
+                            None
+                        } else {
+                            Some(group_members)
+                        };
+
+                        let group_admins_result: Result<Vec<requests::GroupAdmin>, _> = admin_info
                             .into_iter()
-                            .map(|(id, user_pub_key)| {
-                                let maybe_transform_key = recrypt.generate_transform_key(
-                                    &group_priv_key.clone().into(),
-                                    &user_pub_key.clone().into(),
+                            .map(|(admin_id, admin_pub_key)| {
+                                let encrypted_group_key = recrypt.encrypt(
+                                    &plaintext,
+                                    &admin_pub_key.clone().into(),
                                     &auth.signing_private_key().into(),
                                 );
-                                maybe_transform_key.map(|transform_key| {
-                                    (id, (user_pub_key, Some(transform_key.into())))
-                                })
+                                encrypted_group_key
+                                    .map_err(|e| e.into())
+                                    .and_then(EncryptedOnceValue::try_from)
+                                    .map(|enc_msg| requests::GroupAdmin {
+                                        encrypted_msg: enc_msg,
+                                        user: requests::User {
+                                            user_id: admin_id,
+                                            user_master_public_key: admin_pub_key.into(),
+                                        },
+                                    })
                             })
                             .collect();
-                        let member_info = member_info_result?;
+                        let group_admins = group_admins_result?;
```

Comment:
Can this be combined onto the previous statement?

Reply:
It can't (as far as I've tried) because the `.collect()` uses the type I've told it to know what to collect it into, which would then be a different type after the `?`. So it seems to get confused if I combine them.

Reply:
Okay, never mind. In a completely unrelated comment about turbofish, @skeet70 made me realize that I can do this in one line. So I did that.